src/transformers/trainer.py

```diff
@@ -3831,6 +3831,19 @@ def evaluation_loop(
             if is_torch_xla_available():
                 xm.mark_step()
 
+            if self.args.eval_use_gather_object:
+                if "use_gather_object" in inspect.signature(
+                    self.gather_function
+                ).parameters.keys() and is_accelerate_available("0.30.0"):
+                    self.gather_function = functools.partial(
+                        self.gather_function, use_gather_object=self.args.eval_use_gather_object
+                    )
+                else:
+                    logger.warning(
+                        "You are using eval_use_gather_object with a version of `accelerate` < 0.30.0."
+                        "It is recommended to update your version. Since use_gather_object might not be in gather function."
+                    )
+
```

Comment:
Small revision for better readability 

```suggestion
            if self.args.eval_use_gather_object and not is_accelerate_available("0.30.0"):
                logger.warning("You are using eval_use_gather_object = True with a version of `accelerate` < 0.30.0. This is not supported"
                                "and we recommend you to update your version.")
            if "use_gather_object" in inspect.signature(self.gather_function).parameters.keys():
                self.gather_function = functools.partial(
                                self.gather_function, use_gather_object=self.args.eval_use_gather_object
                            )

```